Sources/PitchSpeller/Data Structures/Graph.swift

```diff
@@ -184,41 +214,52 @@ public struct Graph <Value: Hashable>: Hashable {
     }
 
     /// - Returns: A `Graph` with each of the nodes updated by the given `transform`.
-    public func mapNodes <U> (_ transform: (Value) -> U) -> Graph<U> {
-        return Graph<U>(
-            Dictionary(
-                adjacencyList.map { (node,edges) in
-                    (transform(node), edges.map { $0.mapNodes(transform)})
-                }
-            )
+    public func mapNodes <U> (_ transform: (Node) -> U) -> Graph<U, Weight> {
```

Comment:
Looks like you had an argument with the compiler :). I will pull this down to see if I can find the thing the compiler didn't like.

Reply:
Whoops! Yeah, let me know if you can glean anything - thanks :) :)

Reply:
This works for me:

```Swift
return Graph<U,Weight>(
    nodeSet.map(transform),
    Dictionary(
        adjacencyMatrix.map { nodes, edge in
            return (
                Graph<U,Weight>.NodePair(transform(nodes.source),transform(nodes.destination)),
                edge.mapNodes(transform)
            )
        }
    )
)
```

Reply:
I wonder if we are doing extra work here (mapping over the nodes of `nodeSet`, as well as within each `edge`).

Reply:
Yeah, the only reason I have the set of nodes is to cover fully connected nodes, but it does feel a bit redundant. Then again, hard for me to see another way as long as `Node`s are value types. Is this an important function?

Reply:
Ahh, right. Of course. Let's keep this as is.